pkg/resourceinterpreter/defaultinterpreter/prune/prune.go

```diff
@@ -119,3 +123,26 @@ func removeGenerateSelectorOfJob(workload *unstructured.Unstructured) error {
 	}
 	return nil
 }
+
+// RemoveJobTTLSecondsAfterFinished Remove the TTLSecondsAfterFinished filed of the job.
+func RemoveJobTTLSecondsAfterFinished(workload *unstructured.Unstructured) {
+	/*
+	   +--------+------------------------------------------+-------------------------------------------------+----------+
+	   | number | ttl-after-finished controller in karmada | ttl-after-finished controller in member cluster | conflict |
+	   +--------+------------------------------------------+-------------------------------------------------+----------+
+	   | 1      | disable                                  | disable                                         | No       |
+	   | 2      | disable                                  | enable                                          | Yes      |
+	   | 3      | enable                                   | disable                                         | No       |
+	   | 4      | enable                                   | enable                                          | Yes      |
+	   +--------+------------------------------------------+-------------------------------------------------+----------+
+
+	   In the case of number 2, jobs propagated to member clusters will always be recreated once TTL after finished
+	   In the case of number 4, if the member cluster is earlier than karamda's deletion of the completed job, the job of the member cluster will be recreated
+	*/
+	// For the above reasons,ttl-after-finished controller should always be in karmada, not in the member cluster
+	// More details please refer to: https://github.com/karmada-io/karmada/issues/2197.
+	// If you want to enable ttl-after-finished controller, please refer to https://github.com/karmada-io/karmada/blob/master/docs/userguide/configure-controllers.md#optinal-controllers.
+	if workload.GetKind() == util.JobKind {
+		unstructured.RemoveNestedField(workload.Object, "spec", "ttlSecondsAfterFinished")
+	}
+}
```

Comment:
```suggestion
// RemoveJobTTLSeconds removes the '.spec.ttlSecondsAfterFinished' from a Job.
// The reason for removing it is because the Job propagated by Karmada probably be automatically deleted
// from member clusters(by 'ttl-after-finished' controller in member clusters). That will cause a conflict if
// Karmada tries to re-create it. See https://github.com/karmada-io/karmada/issues/2197 for more details.
//
// It is recommended to enable the `ttl-after-finished` controller in the Karmada control plane.
// See https://karmada.io/docs/administrator/configuration/configure-controllers#ttl-after-finished for more details.
func RemoveJobTTLSeconds(workload *unstructured.Unstructured) {
	if workload.GetKind() == util.JobKind {
		unstructured.RemoveNestedField(workload.Object, "spec", "ttlSecondsAfterFinished")
	}
}
```

I suppose we can give it a shorter name.
By the way, I think we can make some additional edits to 
https://karmada.io/docs/administrator/configuration/configure-controllers#ttl-after-finished.